apiserver/admin.go

```diff
@@ -453,16 +456,30 @@ func (a *admin) checkUserPermissions(authInfo authentication.AuthInfo, controlle
 	// everyone group.
 	everyoneGroupAccess := permission.NoAccess
 	if !userTag.IsLocal() {
-		everyoneTag := names.NewUserTag(common.EveryoneTagName)
-		everyoneGroupUser, err := state.ControllerAccess(a.root.state, everyoneTag)
-		if err != nil && !errors.Is(err, errors.NotFound) {
+
+		// TODO (manadart 2024-05-27): This is a huge wart.
+		// Having to acquire the service this way is an abysmal failure of
+		// encapsulation.
+		// The best option might be to round up all of these special case
+		// "everyone@external" checks and push them into either the permission
+		// delegator or the service itself.
+		accessService := a.srv.shared.serviceFactoryGetter.FactoryForModel(database.ControllerNS).Access()
+		everyoneGroupUser, err := accessService.ReadUserAccessForTarget(
+			context.TODO(),
+			common.EveryoneTagName,
```

Comment:
There will be a Jira task to push this check into the service, so it's a fall-through for checks by default.